app/src/main/java/com/folen/androidshowreel/activity/MainActivity.java

```diff
@@ -56,18 +57,38 @@ private void init() {
     private void setupCheckboxes() {
         checkboxAll = findViewById(R.id.checkbox_all);
         checkboxDone = findViewById(R.id.checkbox_done);
-        checkboxTODO = findViewById(R.id.checkbox_todo);
+        checkboxTodo = findViewById(R.id.checkbox_todo);
+
+        View.OnClickListener onClickListener = new View.OnClickListener() {
+            @Override
+            public void onClick(View v) {
+                if (v == checkboxAll) {
+                    checkboxDone.setChecked(false);
+                    checkboxTodo.setChecked(false);
+                    itemAdapter.filter("all");
+                }
+                if (v == checkboxDone) {
+                    checkboxAll.setChecked(false);
+                    checkboxTodo.setChecked(false);
+                    itemAdapter.filter("done");
+                }
+                if (v == checkboxTodo) {
+                    checkboxAll.setChecked(false);
+                    checkboxDone.setChecked(false);
+                    itemAdapter.filter("todo");
+                }
+            }
+        };
 
-        checkboxAll.setOnClickListener(this);
-        checkboxDone.setOnClickListener(this);
-        checkboxTODO.setOnClickListener(this);
+        checkboxAll.setOnClickListener(onClickListener);
```

Comment:
To tak z glowy bez sprawdzenia wiec sprawdz czy wszystko jest dobrze - chce zeby cala logika byla tu wypisane czyli ze takie przypadki jak to ze zaznaczasz sobie DONE a TODO juz miales to znaczy ze checmy all wiec filtrujemy po all, a odznaczamy te dwa checkboxy i zaznaczamy all
```
checkboxAll.setOnCheckedChangeListener((buttonView, isChecked) -> {
      checkboxDone.setChecked(false);
      checkboxTodo.setChecked(false);
    });
    checkboxDone.setOnCheckedChangeListener((buttonView, isChecked) -> {
      if (checkboxTodo.isChecked() && isChecked) {
        checkboxAll.setChecked(true);
        checkboxDone.setChecked(false);
        checkboxTodo.setChecked(false);
        filterListBy(FeatureRealization.ALL);
      } else {
        checkboxAll.setChecked(false);
        filterListBy(FeatureRealization.DONE);
      }
    });
    checkboxTodo.setOnCheckedChangeListener((buttonView, isChecked) -> {
      if (checkboxDone.isChecked() && isChecked) {
        checkboxAll.setChecked(true);
        checkboxDone.setChecked(false);
        checkboxTodo.setChecked(false);
        filterListBy(FeatureRealization.ALL);
      } else {
        checkboxAll.setChecked(false);
        filterListBy(FeatureRealization.TODO);
      }
    });
```